PowerEditor/src/Notepad_plus.cpp

```diff
@@ -2615,18 +2617,23 @@ void Notepad_plus::checkDocState()
 	if (_pAnsiCharPanel)
 		_pAnsiCharPanel->switchEncoding();
 
-	enableCommand(IDM_VIEW_MONITORING, !curBuf->isUntitled(), MENU | TOOLBAR);
-	checkMenuItem(IDM_VIEW_MONITORING, curBuf->isMonitoringOn());
-	_toolBar.setCheck(IDM_VIEW_MONITORING, curBuf->isMonitoringOn());
+	isCurrentDirty = curBuf->isDirty();
+	isCurrentUntitled = curBuf->isUntitled();
+	isCurrentMonitoringOn = curBuf->isMonitoringOn();
+	isCurrentInaccessible = curBuf->isInaccessible();
+
+	enableCommand(IDM_VIEW_MONITORING, !isCurrentUntitled, MENU | TOOLBAR);
+	checkMenuItem(IDM_VIEW_MONITORING, isCurrentMonitoringOn);
+	_toolBar.setCheck(IDM_VIEW_MONITORING, isCurrentMonitoringOn);
 
-	enableCommand(IDM_FILE_SAVEAS, !curBuf->isInaccessible(), MENU);
-	enableCommand(IDM_FILE_RENAME, !curBuf->isInaccessible(), MENU);
-	if (curBuf->isInaccessible())
-		enableCommand(IDM_EDIT_CLEARREADONLY, false, MENU);
-	enableCommand(IDM_VIEW_GOTO_ANOTHER_VIEW, !curBuf->isInaccessible(), MENU);
-	enableCommand(IDM_VIEW_CLONE_TO_ANOTHER_VIEW, !curBuf->isInaccessible(), MENU);
-	enableCommand(IDM_VIEW_GOTO_NEW_INSTANCE, !curBuf->isInaccessible() && !curBuf->isDirty() && !curBuf->isUntitled(), MENU);
-	enableCommand(IDM_VIEW_LOAD_IN_NEW_INSTANCE, !curBuf->isInaccessible() && !curBuf->isDirty() && !curBuf->isUntitled(), MENU);
+	enableCommand(IDM_FILE_SAVEAS, !isCurrentInaccessible, MENU);
+	enableCommand(IDM_FILE_RENAME, !isCurrentInaccessible, MENU);
+	enableCommand(IDM_VIEW_GOTO_ANOTHER_VIEW, !isCurrentInaccessible, MENU);
+	enableCommand(IDM_EDIT_TOGGLE_READONLY_SYS, !isCurrentInaccessible && isCurrentFileExisting && !isCurrentUntitled && !isCurrentMonitoringOn && !isCurrentDirty, MENU);
+	enableCommand(IDM_VIEW_CLONE_TO_ANOTHER_VIEW, !isCurrentInaccessible, MENU);
+
+	enableCommand(IDM_VIEW_GOTO_NEW_INSTANCE, !isCurrentInaccessible && !isCurrentDirty && !isCurrentUntitled, MENU);
+	enableCommand(IDM_VIEW_LOAD_IN_NEW_INSTANCE, !isCurrentInaccessible && !isCurrentDirty && !isCurrentUntitled, MENU);

```

Comment:
... to this line.

Restore the original order and modify only necessary lines to make code review easily.

Restore the original variable name as well.